### PR TITLE
Support deregistering a chatroom by a room's alias

### PR DESCRIPTION
**Pull request by panpawn**

### CODE REVIEW

File: chat-commands.js [hunk 1]
```diff
@@ -687,7 +687,7 @@ exports.commands = {
 		this.errorReply("NOTE: You probably want to use `/deleteroom` now that it exists.");
 		let id = toId(target);
 		if (!id) return this.parse('/help deregisterchatroom');
-		let targetRoom = Rooms.search(id);
+		let targetRoom = Rooms.search(id) ? Rooms.search(id).id : false;
```
Comment (sirDonovan) [inline chat-commands.js]: This doesn't do what you intend it to (<code>The room 'undefined' isn't registered.</code>). Line 693 is the one you need to change.

Comment (panpawn) [inline chat-commands.js]: Oh, good catch

**Comment by panpawn** [comment]
Updated

**Comment by TheImmortal** [comment]
Deregistering a room isn't something that's done often and it should be done with complete certainly. Allowing aliases leaves the potential for mistakes, such as the wrong room being deregistered. It's not that difficult to type the actual room name and anyone deregistering rooms should be able to.

**Comment by Zarel** [comment]
I agree with TI. Why is this necessary?

**Comment by panpawn** [comment]
I suppose it's not

**Pull request closed without merging**